src/state/nftMarket/helpers.ts

```diff
@@ -974,6 +1051,69 @@ export const combineNftMarketAndMetadata = (
   return completeNftData
 }
 
+const fetchWalletMarketData = async (
+  walletNftsByCollection: { collectionAddress: string; idWithCollectionAddress: TokenIdWithCollectionAddress[] }[],
+): Promise<TokenMarketData[]> => {
+  const walletMarketDataRequests = walletNftsByCollection.map(async (walletNftByCollection) => {
+    const tokenIdIn = walletNftByCollection.idWithCollectionAddress.map((walletNft) => walletNft.tokenId)
+    const nftsOnChainMarketData = await getNftsOnChainMarketData(
+      walletNftByCollection.collectionAddress.toLowerCase(),
+      tokenIdIn,
+    )
+    const nftsMarketData = await getNftsMarketData({
+      tokenId_in: tokenIdIn,
+      collection: walletNftByCollection.collectionAddress.toLowerCase(),
+    })
+
+    return tokenIdIn
+      .map((tokenId) => {
+        const nftMarketData = nftsMarketData.find((tokenMarketData) => tokenMarketData.tokenId === tokenId)
+        const onChainMarketData = nftsOnChainMarketData.find(
+          (onChainTokenMarketData) => onChainTokenMarketData.tokenId === tokenId,
+        )
+
+        if (!nftMarketData && !onChainMarketData) return null
+
+        return { ...nftMarketData, ...onChainMarketData }
+      })
+      .filter((value) => value && Object.keys(value).length)
+  })
+
+  const walletMarketDataResponses = await Promise.all(walletMarketDataRequests)
+  return walletMarketDataResponses.flat()
+}
+
+const fetchMarketDataForSaleNfts = async (collections: ApiCollections, account: string): Promise<TokenMarketData[]> => {
+  try {
+    const onChainForSaleNfts = await getAccountNftsOnChainMarketData(collections, account)
+
+    const onChainForSaleNftsCombinedRequests = Object.entries(
+      groupBy(onChainForSaleNfts, (onChainTokenMarketData) => onChainTokenMarketData.collection.id),
+    ).map(async ([collectionAddress, onChainTokenMarketDatas]) => {
+      const tokenIds = onChainTokenMarketDatas.map((marketData) => marketData.tokenId)
+      const marketDataForSaleNfts = await getNftsMarketData({
+        tokenId_in: tokenIds,
+        collection: collectionAddress.toLowerCase(),
```

Comment:
`collectionAddress` is already lower case on this line `collection: { id: collectionList[askCallIndex].address.toLowerCase() }`

Reply:
Done